diff --git a/lib/server/configValidation.js b/lib/server/configValidation.js
index cd9a8eb5ede2519841d3fecdf37e46a15a73d05c..18af5c80cf2a13ed5f198a21ea65ffa6e5075cb5 100644
--- a/lib/server/configValidation.js
+++ b/lib/server/configValidation.js
@@ -127,7 +127,7 @@ const SiteUrlSchema = utils_validation_1.Joi.string()
 exports.ConfigSchema = utils_validation_1.Joi.object({
     baseUrl: utils_validation_1.Joi.string()
         .required()
-        .custom((value) => (0, utils_1.addLeadingSlash)((0, utils_1.addTrailingSlash)(value))),
+        .custom((value) => (0, utils_1.addTrailingSlash)(value)),
     baseUrlIssueBanner: utils_validation_1.Joi.boolean().default(exports.DEFAULT_CONFIG.baseUrlIssueBanner),
     favicon: utils_validation_1.Joi.string().optional(),
     title: utils_validation_1.Joi.string().required(),